bip-0325.mediawiki

```diff
@@ -26,36 +26,37 @@ A new type of test network would be more suitable for integration testing by org
-|-
-|Uint32||4||nTime
-|-
-|Uint32||4||nBits
-|}
-
-The <code>modifiedMerkleRoot</code> hash is obtained by generating the merkle root of the block transactions, with the coinbase witness commitment as is, without the signet extension. This means the merkle root of the block is different from the merkle root in the signet commitment. This is needed, because the signature can never be included in the very message (in this case, a block) that is being signed. Apart from the signature, to facilitate block generation (mining), the block nonce value is the only other component of the block that the signet signature does not commit to. When grinding proof of work, the extended nonce cannot be used as it would invalidate the signature. Instead, simply resigning the same (or an updated) block will give a new search space.
-
-A block is considered fully validated if the above commitment is found, and its solution is valid. It is recommended that this verification is done directly before or after the witness commitment verification, as the data required to do both is approximately the same.
+    4 bytes - Signet scriptSig header (0xecc7daa2)
+    x bytes - Solution (scriptSig)
+
+or
+
+    1-4 bytes - Push the following (x + 4) bytes
+    4 bytes - Signet scriptWitness header (0xecc7daa3)
+    x bytes - Solution (scriptWitness)
+
+Any push operations that do not start with either of the 4 byte signet header are ignored. Multiple push operations with either 4 byte signet header are ignored except for the first instance of each header.
+
+To sign the block or verify a block signature, a virtual transaction with a single input and output is constructed from the block as follows:
+
+    nVersion = 1
+    nLockTime = block.nTime
+    vin[0].prevout.hash = block.hashPrevBlock
+    vin[0].prevout.n = 0
+    vin[0].nSequence = 0
+    vout[0].nValue = block.nVersion
+    vout[0].scriptPubKey = signet_merkle_root
+
+The scriptSig and/or scriptWitness for <code>vin[0]</code> are filled in from the secondary commitment push above.
+
+The <code>signet_merkle_root</code> hash is obtained by generating the merkle root of the block transactions, with the coinbase witness commitment as is, without the signet solution (that is the secondary commitment includes a four byte push of the Signet scriptSig header or scriptWitness header, with not additional solution data). This means the merkle root of the block is different from the merkle root in the signet commitment. This is needed, because the signature can never be included in the very message (in this case, a block) that is being signed. Apart from the signature, to facilitate block generation (mining), the block nonce value is the only other component of the block that the signet signature does not commit to. When grinding proof of work, the extended nonce cannot be used as it would invalidate the signature. Instead, simply resigning the same (or an updated) block will give a new search space.
```

Comment:
```suggestion
The <code>signet_merkle_root</code> hash is obtained by generating the merkle root of the block transactions, with the coinbase witness commitment as-is, without the signet solution (that is, the secondary commitment includes a 4-byte push of the Signet scriptSig header or scriptWitness header, without additional solution data). This means the merkle root of the block is different from the merkle root in the signet commitment. This is needed, because the signature can never be included in the very message (in this case, a block) that is being signed. Apart from the signature, to facilitate block generation (mining), the block nonce value is the only other component of the block that the signet signature does not commit to. When grinding proof of work, the extended nonce cannot be used as it would invalidate the signature. Instead, simply resigning the same (or an updated) block will give a new search space.
```